File: server/FinanceTracker.Api/Models/Account.cs
using System.ComponentModel.DataAnnotations;

namespace FinanceTracker.Api.Models
{
    public class Account
    {
        public Guid Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        public AccountType Type { get; set; }

        [Required]
        public decimal Balance { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }
    }

    public enum AccountType
    {
        Checking = 1,
        Savings = 2,
        CreditCard = 3
    }
}

File: client/FinanceTrackerClient/src/app/models/account.ts
export interface Account {
  success: boolean;
  id: string;
  name: string;
  type: AccountType;
  balance: number;
  createdAt: Date;
  updatedAt: Date;
}

export enum AccountType {
  Checking = 1,
  Savings = 2,
  CreditCard = 3
}

export interface AccountRequest {
  name: string;
  type: AccountType;
}

export interface CreateAccountRequest extends AccountRequest {
  balance: number;
}

export interface UpdateAccountRequest extends AccountRequest {
}

export const ACCOUNT_TYPE_OPTIONS = [
  { value: AccountType.Checking, label: 'Checking' },
  { value: AccountType.Savings, label: 'Savings' },
  { value: AccountType.CreditCard, label: 'Credit Card' }
];
